controllers/hosting/endpointconfig_reconciler.go

```diff
@@ -78,7 +78,9 @@ func (r *endpointConfigReconciler) Reconcile(ctx context.Context, desiredDeploym
 		return errors.Wrap(err, "Unable to interpret HostingDeployment endpoint config")
 	}
 
-	r.log.Info("Desired endpoint config", "desired", desiredEndpointConfig)
+	if desiredEndpointConfig != nil {
```

Comment:
if this is ever nil, the job wont be accepted by API server because some of the fields in the endpointconfig would be mandatory
 
even if it gets past that extraction would fail and that would be an error in previous block itself ?

Reply:
I only added this because I was getting panics because of this `log`. This isn't anything to do with this PR, necessarily. I found this is an older PR, this is an ongoing issue.